File: Spam_Ham_Rahul/spam_or_ham.pl
spam_keyword("buy now").
spam_keyword("limited time").
spam_keyword("earn money").
spam_keyword("free gift").
spam_keyword("click here").
spam_keyword("unsubscribe").

suspicious_sender_domain("example.com").
suspicious_sender_domain("spammydomain.com").

lowercase(String, LowercaseString) :-
    string_lower(String, LowercaseString).

contains_spam_keyword(Email) :-
    spam_keyword(Keyword),
    sub_string(Email, _, _, _, Keyword).

contains_suspicious_domain(Email) :-
    suspicious_sender_domain(Domain),
    sub_string(Email, _, _, _, Domain).

is_spam(Email) :-
    lowercase(Email, LowercaseEmail),
    (contains_spam_keyword(LowercaseEmail) ; contains_suspicious_domain(LowercaseEmail)).

email_text("Congratulations! Congratulations!! Congratulations!!!  You've won a free gift. Click here to claim it now.").

:- initialization(main).
main :-
    email_text(EmailText),
    (is_spam(EmailText) -> writeln("This email is likely spam.") ; writeln("This email is not spam.")).

File: Spam_Ham_Rahul/spam_or_ham.py
(defun is-spam (email)
  (setq spam-keywords '("buy now" "limited time" "earn money" "free gift" "click here" "unsubscribe"))
  (setq suspicious-sender-domains '("example.com" "spammydomain.com"))
  
  (setq email (string-downcase email))
  
  (loop for keyword in spam-keywords
        until (not (search keyword email))
        return t)
  
  (loop for domain in suspicious-sender-domains
        until (not (search (format nil "@~a" domain) email))
        return t)
  
  nil)

(let ((email-text "Congratulations! Congratulations!! Congratulations!!! You've won a free gift. Click here to claim it now."))
  (if (is-spam email-text)
      (format t "This email is likely spam.~%")
      (format t "This email is not spam.~%")))
